### ISSUE TITLE
Matcher function only being passed url-string instead of full request or options.

### ISSUE THREAD

**Issue by jwo** [wheresrhys/fetch-mock#237]
The [API docs](https://github.com/wheresrhys/fetch-mock/blob/master/docs/api.md#mockmatcher-response-options-or-mockoptions) state that I should be able to pass a function for the matcher.
 
> Function(url, opts): A function (returning a Boolean) that is passed the url and opts fetch() is called with (or, if fetch() was called with one, the Request instance)

My interpretation of this is that if fetch is called with a `Request` instance, that my function would receive that instance. Instead, I'm receiving the `url` string and no `opts` at all.

What my code does:

```
const request = new Request(url, {
    method: "GET",
    credentials: "include",
    headers: createHeader(),
    mode: "cors"
  });

export async function get(url) {
  const response = await fetch(request);
  const data = await response.json();
  return data;
}
```

I then wanted to test that the headers are correctly being set. Something in test.js like the following.

```js
    fetchMock.mock(
      request => {
        return request.headers["x-token"] === expectedToken;
      },
      { hello: true }
    );
```

Expected: `request` is the full request object
Actual: `request` is only the URL and I have no way to test body or headers

I'm not sure if this is a bug in the code, or in the docs, or even in my interpretation of the docs. I believe the "error" (if we can call it that), stems from https://github.com/wheresrhys/fetch-mock/blob/master/src/compile-route.js#L112 :

```js
if (typeof route.matcher === "function") {
    matchUrl = route.matcher;
} else if { ///... snip
```

Here the matchURL gets set to the function I pass in. But later, it's called with `matchUrl(req, options)`, but `options` is null.

Should we extract the options from the route and pass to `matchUrl`? That would solve my problem specifically.

**Comment by jwo** [comment]
To solve my specific case, I'm probably going to move to a `header` option; but wanted to report and offer to help in case others run into this.

**Comment by wheresrhys** [comment]
Would you mind trying `npm install wheresrhys/fetch-mock#v6` to see if the bug is fixed in the shortly to be released v6 of the library.

Many thanks

**Comment by wheresrhys** [comment]
I'm pretty sure this will happen in v6 too. Not sure of the best way to tackle this flaw - will give it some thought

**Comment by wheresrhys** [comment]
Fixed in v6.0.0-beta.3

**Comment by jwo** [comment]
@wheresrhys I'm sorry I didn't get back to you in time; would you still like me to check in v6 if it was fixed?
